flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/filesystem/FileSystemOutputFormat.java

```diff
@@ -0,0 +1,224 @@
+		this.parameters = parameters;
+	}
+
+	@Override
+	public void open(int taskNumber, int numTasks) throws IOException {
+		try {
+			PartitionTempFileManager fileManager = committer.createTempFileManager(
+					taskNumber, CHECKPOINT_ID);
+			PartitionWriter.Context<T> context = new PartitionWriter.Context<>(
+					parameters, formatFactory);
+			writer = partitionWriterFactory.create(context, fileManager, computer);
+		} catch (Exception e) {
+			throw new TableException("Exception in open", e);
+		}
+	}
+
+	@Override
+	public void writeRecord(T record) throws IOException {
+		try {
+			writer.write(record);
+		} catch (Exception e) {
+			throw new TableException("Exception in writeRecord", e);
+		}
+	}
+
+	@Override
+	public void close() throws IOException {
+		try {
+			writer.close();
+		} catch (Exception e) {
+			throw new TableException("Exception in close", e);
+		}
+	}
+
+	/**
+	 * Builder to build {@link FileSystemOutputFormat}.
+	 */
+	public static class Builder<T> {
+
+		private String[] columnNames;
+		private String[] partitionColumns;
+		private OutputFormatFactory<T> formatFactory;
+		private TableMetaStoreFactory metaStoreFactory;
+		private Path tmpPath;
+
+		private String defaultPartName = "_DEFAULT_PARTITION_";
+		private LinkedHashMap<String, String> staticPartitions = new LinkedHashMap<>();
+		private boolean dynamicGrouped = false;
+		private boolean overwrite = false;
+		private FileSystemFactory fileSystemFactory = FileSystem::get;
+
+		private PartitionComputer<T> computer;
+
+		public Builder<T> setDefaultPartName(String defaultPartName) {
+			this.defaultPartName = defaultPartName;
+			return this;
+		}
+
+		public Builder<T> setColumnNames(String[] columnNames) {
+			this.columnNames = columnNames;
+			return this;
+		}
+
+		public Builder<T> setPartitionColumns(String[] partitionColumns) {
+			this.partitionColumns = partitionColumns;
+			return this;
+		}
+
+		public Builder<T> setStaticPartitions(LinkedHashMap<String, String> staticPartitions) {
+			this.staticPartitions = staticPartitions;
+			return this;
+		}
+
+		public Builder<T> setDynamicGrouped(boolean dynamicGrouped) {
+			this.dynamicGrouped = dynamicGrouped;
+			return this;
+		}
+
+		public Builder<T> setFormatFactory(OutputFormatFactory<T> formatFactory) {
+			this.formatFactory = formatFactory;
+			return this;
+		}
+
+		public Builder<T> setFileSystemFactory(FileSystemFactory fileSystemFactory) {
+			this.fileSystemFactory = fileSystemFactory;
+			return this;
+		}
+
+		public Builder<T> setMetaStoreFactory(TableMetaStoreFactory metaStoreFactory) {
+			this.metaStoreFactory = metaStoreFactory;
+			return this;
+		}
+
+		public Builder<T> setOverwrite(boolean overwrite) {
+			this.overwrite = overwrite;
+			return this;
+		}
+
+		public Builder<T> setTempPath(Path tmpPath) {
+			this.tmpPath = tmpPath;
+			return this;
+		}
+
+		public Builder<T> setPartitionComputer(PartitionComputer<T> computer) {
```

Comment:
never used function

Reply:
In the future we can have `BaseRowPartitionComputer`.